core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala

```diff
@@ -186,7 +186,7 @@ private[spark] class TorrentBroadcast[T: ClassTag](obj: T, id: Long)
               }
               blocks(pid) = new ByteBufferBlockData(b, true)
             case None =>
-              throw new SparkException(s"Failed to get $pieceId of $broadcastId")
+              throw new BlockNotFoundException(pieceId.toString)
```

Comment:
Is it still possible to include the `broadcastId` in the error msg? I feel it's still important information to help debuggability.

Reply:
@jiangxb1987 Actually this is included in the pieceId. 
`val pieceId = BroadcastBlockId(id, "piece" + pid)`

Reply:
okay, should be fine.